bin/util.py

```diff
@@ -568,9 +568,14 @@ def substitute_dir_variables(dirname, variables):
 
 # copies a directory recursively and substitutes {%key%} by their value in text files
 # reference: https://docs.python.org/3/library/shutil.html#copytree-example
-def copytree_and_substitute(src, dst, variables, exist_ok=True, *, preserve_symlinks=True):
+def copytree_and_substitute(src, dst, variables, exist_ok=True, *, preserve_symlinks=True, base=None):
+    if base is None:
+        base = src
+
     if preserve_symlinks and os.path.islink(src):
         shutil.copy(src, dst, follow_symlinks=False)
+    elif os.path.islink(src) and src.resolve().is_relative_to(base):
```

Comment:
currently this is always true.
I don't actually understand why it was broken before anyway: if `preserve_symlinks` is true, then the if-statement above should catch it.

Or is it that for the case in the linked issue, preserve_symlinks should be ignored and symlinks should always be copied as symlinks, even when `preserve_symlinks` is `false`?

Reply:
Yes, it's for the related issue. Symlinks from one file in `skel/problem` to another file in `skel/problem` should always remain symlinks when creating a new problem?

Reply:
yes completely agree